packages/components/src/typography/typography.tsx

```diff
@@ -0,0 +1,124 @@
+import * as tokens from "@chanzuckerberg/czedi-kit-tokens/json/variables.json";
+import React, { ReactNode } from "react";
+import { styled } from "@linaria/react";
+
+type TypographyElement = "h1" | "h2" | "h3" | "h4" | "h5" | "h6" | "span" | "p";
+
+type TypographySize =
+  | "h1"
+  | "h2"
+  | "h3"
+  | "h4"
+  | "h5"
+  | "sm"
+  | "xs"
+  | "default";
+
+// Matches color token categories
+type TypographyColor =
+  | "alert"
+  | "success"
+  | "info"
+  | "warning"
+  | "brand"
+  | "black"
+  | "white";
+
+type TypographyProps = {
+  /**
+   * This prop can be used to specify which size text element should
+   * actually be rendered, in the case that you want to render an element
+   * as one heading but style it as if it were another. If both an `as` prop
+   * and a `size` prop are passed, the `as` will be used to determine the element
+   * and the `size` will be used to determine the classname.
+   */
+  as?: TypographyElement;
+  /**
+   * The text content to present.
+   */
+  children: ReactNode;
+  /**
+   * The color of the text element. If no color provided, a default color
+   * will be used.
+   */
+  color?: TypographyColor;
+  /**
+   * The size of the html element. If no `as` prop is provided, then
+   * the component uses this value to determine which html tag to render
+   * (e.g. 'h1', 'h2', etc.)
+   */
+  size: TypographySize;
+};
+
+const typography = styled.span`
```

Comment:
Linaria does have this `styled` api. Something to think about: when does it make sense to use it over its `css` api?

I like to use `css` when possible, to encourage folks to write non-dynamic CSS. But `styled` is a good option when we definitely need to change styles based on props.

Up to you, though, for what makes sense for the EDS components here 😄